configs/warnings.tslint.json

```diff
@@ -1,15 +1,22 @@
 {
   "rules": {
-    "deprecation": true,
     "await-promise": {
       "severity": "warning",
       "options": [
         "Thenable",
         "PromiseLike"
       ]
     },
+    "deprecation": true,
     "no-return-await": {
       "severity": "warning"
-    }
+    },
+    "typedef": [
+      true,
+      "call-signature",
```

Comment:
it should be only `call-signature` and `property-declaration`, other are not checking return types, but places where we want a type be inferred.